openlibrary/core/models.py

```diff
@@ -236,25 +237,28 @@ def get_ebook_info(self):
                 "borrow_url": "/books/OL1M/foo/borrow",
                 "borrowed": False
             }
+
         """
         d = {}
         if self.ocaid:
             d['has_ebook'] = True
             d['daisy_url'] = self.url('/daisy')
+            d['daisy_only'] = True
 
             collections = self.get_ia_collections()
-            borrowable = ('lendinglibrary' in collections or
-                         ('inlibrary' in collections and inlibrary.get_library() is not None))
+            borrowable = self.can_borrow()
 
             if borrowable:
                 d['borrow_url'] = self.url("/borrow")
                 key = "ebooks" + self.key
                 doc = self._site.store.get(key) or {}
                 d['borrowed'] = doc.get("borrowed") == "true"
+                d['daisy_only'] = False
             elif 'printdisabled' in collections:
-                pass # ebook is not available
+                d['daisy_only'] = True
```

Comment:
Since `d['daisy_only']` is defaulted to True above, seems like this should either be a `pass` or else you could remove the initialization to True. I'd probably favor the former so that we're guaranteed of having some value set, no matter how twisty the conditionals.